src/meltano/core/project_init_service.py

```diff
@@ -43,10 +44,15 @@ def init(self, activate: bool = True, add_discovery: bool = False) -> Project:
         """
         try:
             os.mkdir(self.project_name)
-        except Exception as exp:  # noqa: F841
+        except FileExistsError:
             raise ProjectInitServiceError(
                 f"Directory {self.project_name} already exists."
             )
+        except PermissionError:
+            raise ProjectInitServiceError(
+                f"Permission denied to create {self.project_name}."
+            )
+
```

Comment:
Might be worth catching other, unexpected errors:

```suggestion
        except PermissionError:
            raise ProjectInitServiceError(
                f"Permission denied to create {self.project_name}."
            )
        except Exception as exc:
            raise ProjectInitServiceError(f"Project initialization failed. {exc}")
```